docs/src/content/docs/core/define-options.mdx

```diff
@@ -0,0 +1,105 @@
+---
+title: defineOptions
+description: Allows defining an integration options while keeping the whole thing type-safe.
+---
+
+`defineOptions` allows defining an integration options while keeping the whole thing type-safe.
+
+```ts title="my-integration/index.ts" "defineOptions" ins={3-5,9}
+import { defineIntegration, defineOptions } from "astro-integration-kit";
+
+type Options = {
+    foo?: string | undefined;
+}
+
+export default defineIntegration({
+    // ...
+    options: defineOptions<Options>({ foo: "bar" })
+})
+```
+
+## Usage
+
+Pass an explicit generic type to `defineOptions`. We recommend that you extract it so that:
+
+1. Your users can consume it
+2. You can add JSDoc annotations easily
+
+```ts title="my-integration/index.ts" ins={3-6,10}
+import { defineIntegration, defineOptions } from "astro-integration-kit";
+
+type Options = {
+    /** An insightful comment */
+    foo?: string | undefined;
+}
+
+export default defineIntegration({
+    // ...
+    options: defineOptions<Options>({ foo: "bar" })
+})
+```
+
+`defineOptions` also requires an argument to handle defaults. It will be deep merged
+with the user provided options.
+
+:::caution
+When typing optional properties within your `IntegrationOptions`, make sure to specify `undefined` as a possible value.
+
+```ts "undefined"
+interface IntegrationOptions {
+    foo?: string | undefined;
+}
+```
+
+We recommend that you extend the strictest Astro TSConfig. This will help you create a more stable
+integration across many platforms and environments.
+
+```diff lang="json"
+{
+-  "extends": "astro/tsconfigs/strict"
++  "extends": "astro/tsconfigs/strictest"
+}
+```
+:::
+
+## Runtime validation
+
+`defineOptions` only check your options on type level. If you want runtime validation,
+we recommend that you validate the user input using `zod` inside `setup`:
+
+```ts title="my-integration/index.ts"
+import { defineIntegration, defineOptions } from "astro-integration-kit";
+import { z } from "astro/zod";
+
+const OptionsSchema = z.object({
+    foo: z.union([z.array(z.string()), z.literal(undefined)]).optional()
+})
+
+type Options = {
+    /** An insightful comment */
+    foo?: z.infer<typeof OptionsSchema>["foo"]
+}
+
+
+export default defineIntegration({
+    // ...
+    options: defineOptions<Options>({ foo: undefined }),
+    setup({ options: inputOptions }) {
+        const options = optionsSchema.parse(inputOptions);
+
+        return {}
+    }
+})
+```
+
+:::note
+We may change this in the future and update `defineOptions` argument to let you provide
+a `zod` schema. Reach out if you're interested in this!
+:::
+
+## Why do I need a distinct utility?
+
+Because we need plugins types to be inferred by TypeScript, we can't pass a type for
```

Comment:
```suggestion
Because fuckin typescript that's why
```

Reply:
(not a serious suggestion 👀 )

Reply:
MERGING ASAP

Reply:
I'm tempted to do
```suggestion
Because ~~fuckin typescript that's why~~ we need plugins types to be inferred by TypeScript, we can't pass a type for
```

Reply:
let's do it 😂